### DISCUSSION TITLE
`data.table` syntax  in preconditions

### DISCUSSION THREAD

**Discussion by randrescastaneda** [rstudio/pointblank#282] [Q&A]
Dear all, 

Is it possible to use `data.table` syntax in preconditions?  Two reasons for this. First, I already had written some code checking my data before learning about `pointblank`, but all these tests were written in `data.table` syntax. With the preconditions argument, it would extremely easy to incorporate them into the whole data validation process. Second, when checking big data, being able to use data.table will highly speed up the process. 

I tried something like the code below, but it failed with the following message, `Error in .(un = unique(survey_comparability)) : 
  could not find function "."`

```r
 col_vals_equal(vars(diff),
                 value = 1,
                 preconditions = ~.[, 
                                    .("un" = unique(survey_comparability)),
                                    by = .(country_code, survey_coverage)
                                   ][
                                     order(country_code, un)
                                   ][, 
                                     diff := un - shift(un), 
                                     by = .(country_code) 
                                   ][
                                     !is.na(diff)
                                   ]
  )


```

Thank you so much. 
Best,

**Comment by rich-iannone** [comment]
Sorry for the slow response on this. I’m going to have a closer look at this soon as data.table support is an upcoming requirement and so far it is untested (so problems like this are not surprising to me). 

Could you please file an issue with the full text of this post? The first thing I can do is make preconditions work for data.table objects.

**Comment by randrescastaneda** [comment]
Awesome. I just created the issue. #303. 

Thank you so much. 
best,
